app/src/main/java/org/mozilla/focus/session/SessionManager.kt

```diff
@@ -0,0 +1,227 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.session
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.TextUtils
+import org.mozilla.focus.architecture.NonNullLiveData
+import org.mozilla.focus.architecture.NonNullMutableLiveData
+import org.mozilla.focus.customtabs.CustomTabConfig
+import org.mozilla.focus.shortcut.HomeScreen
+import org.mozilla.focus.utils.SafeIntent
+import org.mozilla.focus.utils.UrlUtils
+import java.util.*
+
+/**
+ * Sessions are managed by this global SessionManager instance.
+ */
+class SessionManager private constructor() {
```

Comment:
This should be an `object` instead of a `class`. With that it's automatically a singleton and the INSTANCE reference can go.